src/modules/vision_target_estimator/Position/python_derivation/KF_unified.py

```diff
@@ -0,0 +1,152 @@
+    used to endorse or promote products derived from this software
+    without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+    OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+    AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+
+File: derivation.py
+Description:
+    Derivation of an error-state EKF based on
+    Sola, Joan. "Quaternion kinematics for the error-state Kalman filter." arXiv preprint arXiv:1711.02508 (2017).
+    The derivation is directly done in discrete-time as this allows us to define the desired type of discretization
+    for each element while defining the equations (easier than a continuous-time derivation followed by a block-wise discretization).
+"""
+
+import argparse
+
+import symforce
+symforce.set_epsilon_to_symbol()
+
+import symforce.symbolic as sf
+from symforce import typing as T
+from symforce import ops
+from symforce.values import Values
+
+import sympy as sp
+from utils.derivation_utils import *
+
+# Initialize parser
+parser = argparse.ArgumentParser()
+
+parser.add_argument("--moving", action='store_true', help="generate equations for moving filter")
+
+# Read arguments from command line
+args = parser.parse_args()
+moving = args.moving
+
+# The state vector is organized in an ordered dictionary
+State = Values(
+    pos_rel=sf.V1(),
+    vel_uav=sf.V1(),
+    bias=sf.V1(),
+    acc_target=sf.V1(),
+    vel_target=sf.V1()
+)
+
+if not moving:
+    del State["acc_target"]
+    del State["vel_target"]
+
+class IdxDof():
+    def __init__(self, idx, dof):
+        self.idx = idx
+        self.dof = dof
+
+# TODO: remove dof, all dofs are one here.
+def BuildTangentStateIndex():
+    tangent_state_index = {}
+    idx = 0
+    for key in State.keys_recursive():
+        dof = State[key].tangent_dim()
+        tangent_state_index[key] = IdxDof(idx, dof)
+        idx += dof
+    return tangent_state_index
+
+tangent_idx = BuildTangentStateIndex()
+
+class VState(sf.Matrix):
+    SHAPE = (State.storage_dim(), 1)
+
+class MState(sf.Matrix):
+    SHAPE = (State.storage_dim(), State.storage_dim())
+
+class VMeas(sf.Matrix):
+    SHAPE = (1, State.storage_dim())
+
+def get_Phi_G(dt: sf.Scalar) -> T.Tuple[sf.Matrix, sf.Matrix]:
+    if moving:
+        Phi = sf.Matrix([
```

Comment:
Could you rather write down the equations of motions using the states directly, please?

Reply:
Done here: https://github.com/PX4/PX4-Autopilot/pull/23726/commits/2a91c0b52c40eb718774cdca572c8f843714fce7. Is this what you meant @bresch?